### PR TITLE
Booking example - Add integration tests

### PR DESCRIPTION
**Pull request by GoranPrime**
#### Overview

> Closes: #813 

#### Checklist

*Implementation*
- [x] Implementation matches ticket acceptance criteria and technical notes
- [x] Manually tested against Acceptance Criteria
- [x] UI checked in Light / Dark mode

*Stability*
- [x] Checked if changes affect any features and verified affected features work as expected
- [x] Added unit tests for new code and verified existing tests work as expected

*Code quality*
- [ ] Updated `CHANGELOG.md`, `README.md` and package versions in `pubspec.yaml`
- [ ] Dependencies are updated to latest versions or new tickets are created if there are breaking changes or deprecations
- [ ] If an unrelated part of the codebase needs to be updated or refactored, create tickets with proposed changes

### CODE REVIEW

File: examples/booking_app/integration_test/test_bundle.dart
Comment (JvnSlv) [inline examples/booking_app/integration_test/test_bundle.dart:1]: The generated files should not be committed to Git

File: examples/booking_app/integration_test/tests/positive/search_page_details_test.dart [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+      await homePage.tapFavoritesPageButton();
+      expect(homePage.isListItemVisible('0-1'), true);
+
+      // unfavorite
+      await homePage.tapSearchPageButton();
+      await homePage.tapHotelListItem('0-1');
+      await detailsPage.tapDetailsFavoriteButton('0-1');
+      await detailsPage.tapDetailsBackButton();
+      expect(homePage.isHotelFavorited(), false);
+      ScaffoldMessenger.of(homePage.context).clearSnackBars();
```
Comment (JvnSlv) [inline examples/booking_app/integration_test/tests/positive/search_page_details_test.dart:46]: Could this function for clearing the snack bar be moved to the corresponding page_object class?

File: examples/booking_app/macos/Flutter/GeneratedPluginRegistrant.swift
Comment (JvnSlv) [inline examples/booking_app/macos/Flutter/GeneratedPluginRegistrant.swift:1]: The generated files should not be committed to Git

**Review by JvnSlv — CHANGES_REQUESTED** [review]

File: examples/booking_app/integration_test/tests/positive/search_page_details_test.dart (hunk 1 above)
Comment (GoranPrime) [reply examples/booking_app/integration_test/tests/positive/search_page_details_test.dart]: Good idea, moved

File: examples/booking_app/macos/Flutter/GeneratedPluginRegistrant.swift
Comment (GoranPrime) [reply examples/booking_app/macos/Flutter/GeneratedPluginRegistrant.swift:1]: Good catch, removed

**Review by JvnSlv — APPROVED** [review]

**Pull request merged**